File: app/src/main/java/com/mobiledevpro/apptemplate/ui/mainscreen/presenter/IMain.java
package com.mobiledevpro.apptemplate.ui.mainscreen.presenter;

import android.app.Activity;

/**
 * Interface for main screen
 * <p>
 * Created by Dmitriy V. Chernysh on 20.10.17.
 * dmitriy.chernysh@gmail.com
 * <p>
 * https://fb.me/mobiledevpro/
 * <p>
 * #MobileDevPro
 */

public interface IMain {
    interface View {
        Activity getActivity();
    }

    interface Presenter {

        void bindView(IMain.View view);

        void unbindView();

        void onStartView();

        void onStopView();
    }
}

File: app/src/main/java/com/mobiledevpro/apptemplate/ui/mainscreen/presenter/IMain.kt
package com.mobiledevpro.apptemplate.ui.mainscreen.presenter

import androidx.annotation.StringRes
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.OnLifecycleEvent

/**
 * Interface for main screen
 *
 * Created by Dmitriy V. Chernysh
 * dmitriy.chernysh@gmail.com
 *
 * https://instagr.am/mobiledevpro/
 *
 * #MobileDevPro
 */

interface IMain {
    interface View {
        fun showToast(@StringRes msg: Int)
    }

    interface Presenter : LifecycleObserver {

        @OnLifecycleEvent(Lifecycle.Event.ON_START)
        fun onStartView()

        @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
        fun onStopView()
    }
}
